strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/PypiArtifactGenerator.java

```diff
@@ -191,10 +208,78 @@ private void createSourcePackageFiles(ZipOutputStream zos,
                                               topLevelPath).getBytes();
         createZipEntry(zos, sourcesPath, sourcesContent);
 
+        copyLicenseFiles(zos);
+
         String randomPath = eggDirectory + "/RANDOM.txt";
         TestFileUtils.generateFile(zos, byteSize, randomPath);
     }
 
+    private String getLicenseName() 
+            throws IOException
+    {
+        if (!ArrayUtils.isEmpty(licenses))
+        {
+            if (licenses.length > 1)
+            {
+                throw new IOException("PyPi doesn't support multiple licenses!");
+            }
+
+            return licenses[0].license().getName();
+        }
+
+        return LicenseType.UNLICENSE.getName();
+    }
+    
+    private String getLicensePath()
+            throws IOException
+    {
+        if (!ArrayUtils.isEmpty(licenses))
+        {
+            if (licenses.length > 1)
+            {
+                throw new IOException("PyPi doesn't support multiple licenses!");
+            }
+
+            return licenses[0].destinationPath();
+        }
+
+        return "LICENSE";
+    }
```

Comment:
Similarly to the above, it looks like this method could be dropped as well.

Reply:
Method dropped.